src/odemis/gui/comp/miccanvas.py

```diff
@@ -1865,6 +1865,21 @@ def on_dbl_click(self, evt):
         evt.Skip()
 
     def setView(self, view, tab_data):
-        # Same as function from super class, but don't show crosshair
         super(FastEMAcquisitionCanvas, self).setView(view, tab_data)
         view.show_crosshair.value = False
+        self._tab_data_model.main.overview_streams.subscribe(self._on_overview_streams)
+
+    def _on_overview_streams(self, _):
+        ovv_streams = self._tab_data_model.main.overview_streams.value.values()
+        tab_streams = self._tab_data_model.streams.value
+        # Remove old streams from view
+        for s in tab_streams:
+            if isinstance(s, FastEMOverviewStream) and s not in ovv_streams:
+                tab_streams.remove(s)
+                self.view.removeStream(s)
+
+        # Add stream to view if it's not already there
+        for s in ovv_streams:
+            if isinstance(s, FastEMOverviewStream) and s not in tab_streams:
+                tab_streams.append(s)
+                self.view.addStream(s)
```

Comment:
It's really, really, not the canvas who should take care of this!
The canvas is simple: it looks at the streams in the view.stream_tree, and show them.
I'd say it's up to the streams controller (or the tab controller) to decide which stream is shown on which view.

Reply:
I have thought quite a lot about where to put it. I know that this is not a controller class, so it shouldn't contain functionality like that. On the other hand, there are other classes in `miccanvas.py` doing things like stage moves, so I thought we don't mind so much about MVC separation here. The advantage is that the same canvas is used for the acquisition and overview tab, so we don't need to duplicate any code. My other idea was to have this code in `viewport.py` and write a superclass for `FastEMAcquisitionViewport` and `FastEMAcquisitionViewport` (with multiple inheritance) to avoid duplication, but in the end I didn't like that too much either. 

Another option is to write a new subclass `FastEMStreamsController` of `StreamBarController` and add it to both FastEM tabs. I think this might be the best option, I'll see if I can implement this.

Reply:
Yes, looks much better now!